contrib/jinzhu/gorm/gorm.go

```diff
@@ -7,17 +7,109 @@
 package gorm
 
 import (
+	"context"
+	"math"
+	"time"
+
 	sqltraced "gopkg.in/DataDog/dd-trace-go.v1/contrib/database/sql"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
 
 	"github.com/jinzhu/gorm"
 )
 
+const (
+	gormContextKey       = "dd-trace-go:context"
+	gormConfigKey        = "dd-trace-go:config"
+	gormSpanStartTimeKey = "dd-trace-go:span"
+)
+
 // Open opens a new (traced) database connection. The used dialect must be formerly registered
 // using (gopkg.in/DataDog/dd-trace-go.v1/contrib/database/sql).Register.
-func Open(dialect, source string) (*gorm.DB, error) {
-	db, err := sqltraced.Open(dialect, source)
+func Open(dialect, source string, opts ...Option) (*gorm.DB, error) {
+	sqldb, err := sqltraced.Open(dialect, source)
 	if err != nil {
 		return nil, err
 	}
-	return gorm.Open(dialect, db)
+	db, err := gorm.Open(dialect, sqldb)
+	if err != nil {
+		return db, err
+	}
+	return WithCallbacks(db, opts...), err
+}
+
+// WithCallbacks registers callbacks to the gorm.DB for tracing.
```

Comment:
This is very vague. Should we be more descriptive about what this function does? I highly recommend looking at `godoc -http=:8080` for this package to make sure it's nice and user friendly and makes sense in terms of usage. Please add a lot more detail on what this does.

Some ideas:
* Add a link to http://jinzhu.me/gorm/development.html#write-plugins
* List all callbacks which are being registered with (so the user can add more if he wishes, perhaps in the future, as part of another option).
* Explain the outcome (e.g. those operations will be traced automatically)

These are just some suggestions, feel free to use your own style and wording and we can iterate together.

Reply:
Thanks, I'll update the doc comment.

Reply:
Great! Resolving!